### PR TITLE
[observable-extension] Update map events data models.

### PR DESCRIPTION
**Pull request by pengdev**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Update map events data models.</changelog>`.

### Summary of changes

This PR updates the map events data models, so they match the gl-native MapEvents.

List of changes:

* add optional `sourceId` and `tileId` in the `MapLoadingErrorEventData` and `OnMapLoadErrorListener`.
* make `response` in `ResourceEventData` optional.
* make `etag`, `modified`, `expires` inside resource request's `Response` optional.
* add `DataSourceType` , `RequestPriority` , `RequestType` ,`ResponseErrorType` , `ResponseSourceType` enum for `resource-request` event which conform to java docs from gl-native.

### CODE REVIEW

**Review by tobrun — APPROVED** [review]

File: app/src/androidTest/java/com/mapbox/maps/testapp/LoadStyleCallbackTest.kt [hunk 1]
```diff
@@ -38,7 +39,12 @@ class LoadStyleCallbackTest {
...
         },
         onMapLoadErrorListener = object : OnMapLoadErrorListener {
-          override fun onMapLoadError(mapLoadErrorType: MapLoadErrorType, message: String) {
+          override fun onMapLoadError(
+            mapLoadErrorType: MapLoadErrorType,
+            message: String,
+            sourceId: String?,
+            tileId: TileID?
+          ) {
             throw AssertionError("Load $mapLoadErrorType failed with $message")
```
Comment (axti) [inline app/src/androidTest/java/com/mapbox/maps/testapp/LoadStyleCallbackTest.kt:48]: Shouldn't the error message be updated as well?

**Review by axti — APPROVED** [review]

**Comment by kiryldz** [comment]
@pengdev any particular ideas how to make sure we did not miss anything?

**Comment by pengdev** [comment]
> @pengdev any particular ideas how to make sure we did not miss anything?

Currently we are dependent on the javadoc in the MapEvents class. Ideally we could generate these classes in long term.

File: sdk/src/main/java/com/mapbox/maps/extension/observable/resourcerequest/eventdata/RequestDataSourceType.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package com.mapbox.maps.extension.observable.resourcerequest.eventdata
+
+import com.google.gson.annotations.SerializedName
+
+/**
+ * Describes data source of request for resource-request event.
+ */
+enum class RequestDataSourceType {
```
Comment (pengdev) [inline sdk/src/main/java/com/mapbox/maps/extension/observable/resourcerequest/eventdata/RequestDataSourceType.kt:8]: nit: `RequestDataSourceType` doesn't belong to request, we could rename it to `DataSourceType`

File: sdk/src/test/java/com/mapbox/maps/StyleObserverTest.kt [hunk 3]
```diff
@@ -99,8 +99,8 @@ class StyleObserverTest {
     val styleObserver = StyleObserver(mockk(relaxed = true), mockk(relaxed = true), 1.0f)
     val errorListener = mockk<OnMapLoadErrorListener>(relaxed = true)
     styleObserver.onNewStyleLoad(mockk(relaxed = true), errorListener)
-    styleObserver.onMapLoadError(MapLoadErrorType.GLYPHS, "foobar")
-    verify { errorListener.onMapLoadError(MapLoadErrorType.GLYPHS, "foobar") }
+    styleObserver.onMapLoadError(MapLoadErrorType.GLYPHS, "foobar", null, null)
```
Comment (kiryldz) [inline sdk/src/test/java/com/mapbox/maps/StyleObserverTest.kt:102]: shouldn't it be better to check non-null values? to make sure we did not mess up smth

File: sdk/src/main/java/com/mapbox/maps/extension/observable/resourcerequest/eventdata/DataSourceType.kt [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+/**
+ * Describes data source of request for resource-request event.
+ */
+enum class DataSourceType {
+  /**
+   * data source as resource-loader.
+   */
+  @SerializedName("resource-loader")
+  RESOURCE_LOADER,
```
[the remark this answers is not in the document]
Comment (ank27) [reply sdk/src/main/java/com/mapbox/maps/extension/observable/resourcerequest/eventdata/DataSourceType.kt:13]: I have used the same format as other enum classes we have. for eg. [MapLoadErrorType](https://github.com/mapbox/mapbox-maps-android/blob/main/sdk-base/src/main/java/com/mapbox/maps/plugin/delegates/listeners/eventdata/MapLoadErrorType.kt)

File: sdk/src/test/java/com/mapbox/maps/StyleObserverTest.kt (hunk 3 above)
Comment (pengdev) [reply sdk/src/test/java/com/mapbox/maps/StyleObserverTest.kt:102]: it doesn't really matter as we just verify the same parameters are called on the listener.

File: sdk/src/main/java/com/mapbox/maps/extension/observable/resourcerequest/eventdata/DataSourceType.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (pengdev) [reply sdk/src/main/java/com/mapbox/maps/extension/observable/resourcerequest/eventdata/DataSourceType.kt:13]: Shouldn't matter as this is auto formatted by IDE 😄

**Review by tobrun — APPROVED** [review]

**Review by kiryldz — APPROVED** [review]

**Review by ank27 — APPROVED** [review]

**Pull request merged**